### PR TITLE
Rework trusted signal fetching

### PR DESCRIPTION
**Pull request by taymonbeal**
- Our escaping and query string behavior are now aligned with (my best understanding of) Chrome's.
- Iteration order of ads and interest groups, and the security implications thereof, are now documented.

Fixes: #69

### CODE REVIEW

File: frame/auction.ts [hunk 1]
```diff
@@ -23,11 +23,20 @@ import { FetchJsonStatus, tryFetchJson } from "./fetch";
  * used has no use for scoring signals, but when more algorithms are supported,
  * this data will be passed to them.
  *
+ * The request's `keys` query parameter consists of all the ads' rendering URLs,
+ * escaped and then joined with unescaped commas, ordered first alphabetically
+ * by interest group name and then by the order given in the
+ * `joinAdInterestGroup` call, with only the first kept in case of duplicates.
+ * This leaks some information about interest groups and when they were joined
+ * to the trusted server, but that's okay, since it's trusted.
```
Comment (jeffkaufman) [inline frame/auction.ts:31]: Why not just sort by renderingUrl?  Then we leak nothing.

(if there's some reason this is impractical, that's probably OK, but the sort seems pretty easy?)

File: frame/auction.ts [hunk 2]
```diff
@@ -23,11 +23,20 @@ import { FetchJsonStatus, tryFetchJson } from "./fetch";
...
+ *
  * @param hostname The hostname of the page where the FLEDGE Shim API is
  * running.
  */
 export async function runAdAuction(
   trustedScoringSignalsUrl: string | null,
+  // This is temporary until trustedBiddingSignalsUrl is added.
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   hostname: string
```
Comment (jeffkaufman) [inline frame/auction.ts:40]: For what it's worth, I think Chrome probably is going to provide `hostname`

**Review by jeffkaufman — APPROVED** [review]

File: frame/auction.ts (hunk 1 above)
Comment (taymonbeal) [reply frame/auction.ts:31]: Filed #104.

File: frame/auction.ts (hunk 2 above)
Comment (taymonbeal) [reply frame/auction.ts:40]: Filed #103.

**Pull request merged**